src/main/java/me/oyurimatheus/nossomercadolivre/products/QuestionController.java

```diff
@@ -0,0 +1,64 @@
+package me.oyurimatheus.nossomercadolivre.products;
+
+import me.oyurimatheus.nossomercadolivre.users.User;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+import static org.springframework.http.ResponseEntity.created;
+import static org.springframework.http.ResponseEntity.notFound;
+
+@RestController
+@RequestMapping("/api/products/{id}/questions")
+class QuestionController {
+
+    private final ProductRepository productRepository;
+    private final QuestionRepository questionRepository;
+    private final ApplicationEventPublisher publisher;
+
+    QuestionController(ProductRepository productRepository,
+                       QuestionRepository questionRepository,
+                       ApplicationEventPublisher publisher) {
+        this.productRepository = productRepository;
+        this.questionRepository = questionRepository;
+        this.publisher = publisher;
+    }
+
+    @PostMapping
+    ResponseEntity<?> askQuestion(@PathVariable("id") UUID id,
+                                  @RequestBody @Valid NewQuestionRequest newQuestion,
+                                  @AuthenticationPrincipal User user,
+                                  UriComponentsBuilder uriBuilder) {
+
+        Optional<Product> possibleProduct = productRepository.findById(id);
+
+        if (possibleProduct.isEmpty()) {
+            return notFound().build();
+        }
+
+        Product product = possibleProduct.get();
+        var question = newQuestion.toQuestion(user, product);
+        questionRepository.save(question);
+
+        publisher.publishEvent(new QuestionEvent(question, uriBuilder));
+
+        var location = URI.create("/api/products/" + id.toString() + "/questions/" + question.getId());
+        List<Question> questions = questionRepository.findByProduct(possibleProduct.get());
+
+        List<QuestionResponse> response = questions.stream()
```

Comment:
A carga intrínseca foi para 7 pontos por conta desse retorno já com todas as novas coleções :). 

Outra coisa importante,  isso só funcionou pq provavelmente os métodos do repository estão com transações habilitadas por default. Então na hora que chama o save ele já fecha a transação. Na listagem consegue enxergar :). 

Ta tudo bem o código, mas é sempre legal ficar de olho no nível de complexidade e em outros detalhes. 7 pontos é a minha sugestão limite para o controller :).

Reply:
Sim sim! E o spring boot jpa utiliza o open session in view por padrão, se não me engano. Se recebesse a lista de questões você acha que ficaria mais simples? Vou testar aqui o/